File: Python/programmers/etc/level2/마법의_엘리베이터.py
def solution(storey):
    if storey < 10 :
        return min(storey, 11-storey)
    left = storey%10
    return min(left+solution(storey//10), 10-left+solution(storey//10+1))


storey = 16
print(solution(storey)) # 6

storey = 2554
print(solution(storey)) # 16

File: Java/programmers/etc/level2/마법의엘리베이터.java
package CodeTest.Java.programmers.etc.level2;

public class 마법의엘리베이터 {
    public int solution(int storey) {
        int answer = 0;
        
        while (storey != 0) {
            int remainder = storey%10;
            
            if (remainder > 5) {
                answer += 10-remainder;
                storey += 10;
            } else if (remainder < 5) {
                answer += remainder;
            } else {
                if ((storey/10)%10 > 4) {
                    storey += 10;
                }
                answer += remainder;
            }
            
            storey /= 10;
        }

        return answer;
    }
}
